### DISCUSSION TITLE
switching to PaesslerAG/jsonpath

### DISCUSSION THREAD

**Discussion by sjlee** [gavv/httpexpect#139] [Ideas]
We love `httpexpect`, and are using it in our project. However, currently `httpexpect` depends on `github.com/yalp/jsonpath` which does not come with a license. From a legal standpoint, this is a big red flag, and we need a solution for us to be able to use `httpexpect`.

Someone raised this issue directly on the `jsonpath` repo (https://github.com/yalp/jsonpath/issues/3), but there has been no movement.

Can `httpexpect` remove the `jsonpath` dependency so that we can use this legally? One possibility might be to use another one like https://github.com/PaesslerAG/jsonpath which is licensed under BSD-3. Thoughts?

**Comment by sjlee** [comment]
I added a PR that addresses this. Could you please review, and merge it if it is acceptable? Thanks!

**Comment by yalp** [comment]
I just added a license to the https://github.com/yalp/jsonpath project. Sorry for the delay...

https://github.com/yalp/jsonpath/issues/3#issuecomment-366633733

**Comment by sjlee** [comment]
Thanks @yalp! Greatly appreciated.

@gavv, what are your thoughts on proceeding? It sounds like @yalp is indicating one should use https://github.com/PaesslerAG/jsonpath going forward (https://github.com/yalp/jsonpath/blob/master/README.md).

**Comment by gavv** [comment]
We could switch but we should solve the compatibility issues first.. See my comment at #49.
